lib/Service/SynchronizationService.php

```diff
@@ -246,14 +252,19 @@ public function synchronizeContract(SynchronizationContract $synchronizationCont
         // Let's prevent pointless updates @todo account for omnidirectional sync, unless the config has been updated since last check then we do want to rebuild and check if the tagert object has changed
         if ($originHash === $synchronizationContract->getOriginHash() && $synchronization->getUpdated() < $synchronizationContract->getSourceLastChecked()) {
             // The object has not changed and the config has not been updated since last check
-            // return $synchronizationContract;
-            // @todo: somehow this always returns true, so we never do the updateTarget
+			return $synchronizationContract;
         }
 
         // The object has changed, oke let do mappig and bla die bla
         $synchronizationContract->setOriginHash($originHash);
         $synchronizationContract->setSourceLastChanged(new DateTime());
 
+		// Check if object adheres to conditions.
+		// Take note, JsonLogic::apply() returns a range of return types, so checking it with '=== false' or '!== true' does not work properly.
+		if ($synchronization->getConditions() !== [] && !JsonLogic::apply($synchronization->getConditions(), $object)) {
```

Comment:
=== false

Reply:
Did you read the comment above this line? === false is not going to cut it.